src/init/AppRunner.ts

```diff
@@ -127,6 +129,41 @@ export class AppRunner {
     }
   }
 
+  /**
+   * Retrieves settings from package.json or configuration file when
+   * part of an npm project
+   * @returns The settings defined  in the configuration file
+   */
+  public async getPackageSettings(): Promise<undefined | Record<string, unknown>> {
```

Comment:
This is just me preferring certain order of functions, but since there are some other things that need to be fixed anyway: can you put this function definition below `createCli` (and thus above `createComponentsManager`)?